lib/core/macros.nim

```diff
@@ -1545,11 +1545,15 @@ proc customPragmaNode(n: NimNode): NimNode =
               let varNode = identDefs[i]
               # if it is and empty branch, skip
               if varNode[0].kind == nnkNilLit: continue
-              if varNode[1].kind == nnkIdentDefs:
+
+              if varNode.len > 1 and varNode[1].kind == nnkIdentDefs:
                 identDefsStack.add(varNode[1])
               else: # nnkRecList
-                for j in 0 ..< varNode[1].len:
-                  identDefsStack.add(varNode[1][j])
+                var idx: int = 1
+                if varNode.kind == nnkElse:
+                  idx = 0
+                for j in 0 ..< varNode[idx].len:
+                  identDefsStack.add(varNode[idx][j])
```

Comment:
I think it's better to rewrite this piece using a case statement to dispatch between node types and move nnkRecList handling outside since the current version is too hard to follow, something like this perhaps:

```nim
case identDefs.kind
of nnkRecList:
  for child in identDefs.items:
    identDefsStack.add(child)
of nnkRecCase:
  # Add condition definition
  identDefsStack.add(identDefs[0])
  # Add branches
  for i in 1 ..< identDefs.len:
    let varNode = identDefs[i]
    case varNode.kind
    of nnkOfBranch:
      # We are only interested in variable definitions
      if varNode[1].kind in {nnkIdentDefs, nnkRecList}:
        identDefsStack.add(varNode[1])
    of nnkElse:
      # We are only interested in variable definitions
      if varNode[0].kind in {nnkIdentDefs, nnkRecList}:
        identDefsStack.add(varNode[0])
    else:
      doAssert false, "Unknown node kind in nnkRecCase: " & $varNode.kind
else:
  <insert identdefs handling code here>
```

Though the best thing to do would be to remove this mess that is `identDefsStack` and use a recursive function, but that can be left for later.

Reply:
I've replaced the fix with your cleaner version, with small changes.
I'm calling the `children` instead of the `items` iterator and added brackets to the `doAssert` call. I think it's better to read.